project/src/main/webapp/message.js

```diff
@@ -0,0 +1,54 @@
+
+var currentMessages = [];
+
+// Display user's favorite restaurants in "Favorites" bar
+function displayMessageChain() {
+	fetch(`/chat?recipient=${currentStore}`).then(response => response.json()).then((messages) => {
+		var messageChain = document.getElementById('message-chain');
+		var pos = 0;
+		var test = true;
+		for (const msgPosition in messages) {
+			const msg = messages[msgPosition];
+			if (currentMessages.length <= pos && document.getElementById(`msg${pos}`) == null) {
+				var messageElement;
+				messageElement = document.createElement("li");
+				messageElement.id = `msg${pos}`
+				messageElement.innerText = msg.message + " from " + msg.writer;
+				messageChain.appendChild(messageElement);
+				currentMessages.push(msg);
+			} else if (currentMessages[pos] == msg) {
+				continue;
```

Comment:
put this as the first check.

if (currMsgs[pos] == msg) {
  continue;
}

if (curMsg.length < ...) {
} else {
}